### PR TITLE
doc: Handle exception in parsing parameter description

### LINKED ISSUE #60587: ansible-doc failing on specific line in documentation
##### SUMMARY

anisble-doc is failing on below line
https://github.com/dynatrace-innovationlab/ansible_dynatrace_problem_comment/blob/master/library/dynatrace_comment.py#L47
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

ansible-doc
##### ANSIBLE VERSION

```paste below

```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

ansible-doc changes the line to something like below
u'Source where the comment originates from (default': u'Ansible)'
The above is not a valid python string and hence it throws an exception

```yaml

```

##### EXPECTED RESULTS

No exception  should be thrown, instead ansible-doc should fail the test

##### ACTUAL RESULTS

ansible-doc -t module dynatrace_innovationlab.dynatrace_collection.dynatrace_comment dynatrace_innovationlab.dynatrace_collection.dynatrace_deployment -vvv

```paste below
Traceback (most recent call last):
  File "/home/abehl/work/src/anshul_ansible/ansible/bin/ansible-doc", line 111, in <module>
    exit_code = cli.run()
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 198, in run
    textret = DocCLI.format_plugin_doc(plugin, loader, plugin_type, search_paths)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 309, in format_plugin_doc
    text += DocCLI.get_man_text(doc)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 598, in get_man_text
    DocCLI.add_fields(text, doc.pop('options'), limit, opt_indent)
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/doc.py", line 476, in add_fields
    text.append(textwrap.fill(DocCLI.tty_ify(entry), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
  File "/home/abehl/work/src/anshul_ansible/ansible/lib/ansible/cli/__init__.py", line 426, in tty_ify
    t = cls._ITALIC.sub("`" + r"\1" + "'", text)    # I(word) => `word'
TypeError: expected string or buffer
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #60587

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/60587-doc_parsing.yml
lib/ansible/cli/doc.py

### CODE REVIEW

File: lib/ansible/cli/doc.py [hunk 1]
```diff
@@ -473,6 +473,8 @@ def add_fields(text, fields, limit, opt_indent):
 
             if isinstance(opt['description'], list):
                 for entry in opt['description']:
+                    if not isinstance(entry, string_types):
+                        raise AnsibleError("Expected string in %s description, got %s" % (o, type(entry)))
                     text.append(textwrap.fill(DocCLI.tty_ify(entry), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
             else:
```
Comment (bcoca) [inline lib/ansible/cli/doc.py:479]: should have elif/else with same error on 'non list'

File: lib/ansible/cli/doc.py [hunk 2]
```diff
@@ -473,8 +473,12 @@ def add_fields(text, fields, limit, opt_indent):
 
             if isinstance(opt['description'], list):
                 for entry in opt['description']:
+                    if not isinstance(entry, string_types):
+                        raise AnsibleError("Expected string in %s description, got %s" % (o, type(entry)))
```
Comment (bcoca) [inline lib/ansible/cli/doc.py]: since it is a list, you can point to the specific element in description that is a problem

Comment (Akasurde) [reply lib/ansible/cli/doc.py]: `o` contains name of the element which has a problem.

**Review by Andersson007 — APPROVED** [review]
i'd like to add unit tests for this case and in general for doc.py DocCLI class after this is merged.

shipit

File: lib/ansible/cli/doc.py (hunk 2 above)
Comment (bcoca) [reply lib/ansible/cli/doc.py]: its a list of elements, i mean the index of that list:
```
option:
  description:
    - one ...
    - two ...
    - three: is the error
```

**Pull request merged**